ViewLayer/CompactUI/CompactUI.cpp

```diff
@@ -0,0 +1,582 @@
+#include "CompactUI.h"
+#include "ui_CompactUI.h"
+
+CompactUI::CompactUI()
+    :ui(new Ui::CompactUI)
+{
+    mppt1CurrentIn=0;
```

Comment:
I am still trying to understand the full scope of the changes so I might retract this. But generally a UI class should not double up as an entity. My first instinct would be that if you need to store these values within the data layer.